site/docs/hive.md

```diff
@@ -17,113 +17,211 @@
+You can use other compute engines or the Java/Python API to create such a table. 
+
+Suppose there is a table `table_a` and the table location is `hdfs://some_path/table_a`. 
+Now overlay a Hive table on top of this Iceberg table by issuing Hive DDL like so:
+
 ```sql
-SELECT * from table_b;
+CREATE EXTERNAL TABLE table_a 
+STORED BY 'org.apache.iceberg.mr.hive.HiveIcebergStorageHandler' 
+LOCATION 'hdfs://some_bucket/some_path/table_a';
 ```
 
-#### Using Hadoop Catalog
-Iceberg tables created using `HadoopCatalog` are stored entirely in a directory in a filesystem like HDFS.
+#### Hadoop catalog tables
+
+Iceberg tables created using `HadoopCatalog` are stored entirely in a directory in a filesystem like HDFS, 
+similar to the tables created through `HadoopTables`.
+You can use other compute engines or Java/Python API to create such as table. 
 
-##### Create an Iceberg table
-The first step is to create an Iceberg table using the Spark/Java/Python API and `HadoopCatalog`. For the purposes of this documentation we will assume that the fully qualified table identifier is `database_a.table_c` and that the Hadoop Catalog warehouse location is `hdfs://some_bucket/path_to_hadoop_warehouse`. Iceberg will therefore create the table at the location `hdfs://some_bucket/path_to_hadoop_warehouse/database_a/table_c`.
+Suppose there is a table `table_b` and the table location is `hdfs://some_path/table_b`. 
 
-##### Create a Hive table
 Now overlay a Hive table on top of this Iceberg table by issuing Hive DDL like so:
+
 ```sql
-CREATE EXTERNAL TABLE database_a.table_c 
+CREATE EXTERNAL TABLE database_a.table_b
 STORED BY 'org.apache.iceberg.mr.hive.HiveIcebergStorageHandler' 
-LOCATION 'hdfs://some_bucket/path_to_hadoop_warehouse/database_a/table_c'
+LOCATION 'hdfs://some_path/table_b'
 TBLPROPERTIES (
   'iceberg.mr.catalog'='hadoop', 
-  'iceberg.mr.catalog.hadoop.warehouse.location'='hdfs://some_bucket/path_to_hadoop_warehouse')
-;
+  'iceberg.mr.catalog.hadoop.warehouse.location'='hdfs://some_bucket/path_to_hadoop_warehouse'
+);
 ```
+
 Note that the Hive database and table name *must* match the values used in the Iceberg `TableIdentifier` when the table was created. 
 
-It is possible to omit either or both of the table properties but instead you will then need to set these when reading from the table. Generally it is recommended to set them at table creation time so you can query tables created by different catalogs. 
+It is possible to omit either or both of the table properties but instead you will then need to set these when reading from the table.
+Generally it is recommended to set them at table creation time, so you can query tables created by different catalogs. 
+
+#### Hive catalog tables
+
+As described before, tables created by the `HiveCatalog` with Hive engine feature enabled are directly visible by the Hive engine, so there is no need to create an overlay.
+
```

Comment:
Here we should mention that for HiveCatalog we can now create tables using Hive's own columns and partition spec syntax.
```
CREATE TABLE database_a.table_b (id bigint, name string) STORED BY 'org.apache.iceberg.mr.hive.HiveIcebergStorageHandler' PARTITIONED BY (dept string);
```
Note: This creates an unpartitioned HMS table (while the underlying Iceberg table is partitioned).

Also, we could mention the compatibility matrix between Iceberg and Hive types: 

- which Hive types are supported out-of-the-box since they have a direct equivalent in Iceberg
- which Hive types can be autoconverted (e.g. short -> int), if enabled
- which ones are unsupported altogether (e.g. interval).

I'm happy to address this in a follow-up PR though.

Reply:
thanks for the suggestion, I completely forgot this one, added now

Reply:
Thanks for updating it!

Reply:
We should probably make this a table eventually, like we have in the Spark docs.

Reply:
@rdblue added a section in the end for forward and backward conversion.